From: Adam Jackson <ajax@nwnk.net>
Date: Tue, 9 Oct 2012 14:08:24 -0400
Subject: [PATCH] 0.108: PolkitAgent: Avoid crashing if initializing the server
 object fails

Note that otherwise we return a freed server object.  Since later in
polkit_agent_listener_register_with_options we check against NULL to
determine failure, this makes for sad times later when we call
server_free() on it again.

Signed-off-by: David Zeuthen <zeuthen@gmail.com>
Origin: 0.108, commit:59f2d96ce3ac63173669f299a9453a7bf5e70a70
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=55776
Bug-Debian: https://bugs.debian.org/923046

Imported from policykit-1_0.105-31.1.debian.tar.xz

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 src/polkitagent/polkitagentlistener.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/polkitagent/polkitagentlistener.c b/src/polkitagent/polkitagentlistener.c
index 0d97501a028c..5bddd035cc0e 100644
--- a/src/polkitagent/polkitagentlistener.c
+++ b/src/polkitagent/polkitagentlistener.c
@@ -260,10 +260,9 @@ server_new (PolkitSubject  *subject,
   if (!server_init_sync (server, cancellable, error))
     {
       server_free (server);
-      goto out;
+      return NULL;
     }
 
- out:
   return server;
 }
 
